### PR TITLE
Fixed content list css bug and full details column

### LINKED ISSUE #56: Fields should not cross over two columns
The group dropdown is cut in half as long as the input field stays on the left side.

If you have enough input fields on the left, like adress and the group input goes to right it's ok again.

**Nextcloud version:** 11 RC1
**Contacts version:** 1.5.2

chrome_2016-12-08_07-59-20

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/39846542-group-dropdown-misplaced?utm_campaign=plugin&utm_content=tracker%2F46751899&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F46751899&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #453: NC13b3 regression: big vertical spaces in left column
### Steps to reproduce
1. Upgrade to Nextcloud 13 (b3 in my case)
2. Open Contacts app
3. Group chooser / Left column is no longer neatly formatted

### Expected behaviour
Should look nicely, as it did in NC 12.
### Actual behaviour
Large vertical spaces between lines:
grafik

### Server configuration
**Operating system**: Debian Stretch (up to date)
**Web server:** Apache 2.4.10
**Database:** MySQL 5.5.58
**PHP version:** 5.6.30
**Nextcloud version:** 13.0.0 Beta 3
**Updated from an older Nextcloud/ownCloud or fresh install:** Upgraded from 12.0.4
**Where did you install Nextcloud from:** admin page / installer
**Signing status:** no errors upon signing: http://example.com/index.php/settings/integrity/failed reports "No errors have been found."
**Contacts version:** 2.0.1
**Where did you install Contacts from:** apps page / updated after NC upgrade

**List of activated apps:**: Seems irrelevant, isn't it?
**Nextcloud configuration:**
<details>
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "seyfarth.de"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/seyfarth.de\/cloud",
        "dbtype": "mysql",
        "version": "13.0.0.8",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "Europe\/Berlin",
        "installed": true,
        "trashbin_retention_obligation": "auto,90",
        "versions_retention_obligation": "90,auto",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": "true",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379,
            "timeout": 0
        },
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "php",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "defaultapp": "calendar",
        "maintenance": false,
        "loglevel": 2,
        "theme": "",
        "appstore.experimental.enabled": true,
        "updater.release.channel": "beta",
        "mail_smtpauthtype": "PLAIN",
        "mail_smtpsecure": "tls",
        "mail_smtpauth": 1
    }
}
</details>

**Are you using external storage, if yes which one:** no
**Are you using encryption:** no
**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Firefox 57 / Chrome 63
**Operating system:** W10
**CardDAV-clients:**: issue ist with NC web interface only (browser)

#### Web server error log: No entry while doing password reset.
But I see this:
[ssl:error] AH02032: Hostname www.seyfarth.de provided via SNI and hostname seyfarth.de provided via HTTP are different
Yet, Let's Encrypt seems to be correct:
Certificate Name: seyfarth.de
  Domains: www.seyfarth.de seyfarth.de
  Expiry Date: 2018-03-07 22:32:14+00:00 (VALID: 78 days)
  Paths: ...
#### Browser log (Firefox Browser Console):
JQMIGRATE: Migrate is installed, version 1.4.0  core.js:7
getAttributeNode() sollte nicht mehr verwendet werden. Verwenden Sie stattdessen getAttribute().  core.js:7:1714
Das nodeValue-Attribut von Attributen sollte nicht mehr verwendet werden. Verwenden Sie value stattdessen.  core.js:7:1714
Content Security Policy: Die Einstellungen der Seite haben das Laden einer Ressource auf self blockiert ("script-src 'nonce-NGRPZTRSdW45empjcHF3YUUwK0MwUXE3a1VrS3lWV2Fua3YrRXJOTldLUT06cTc2cTJFM2pyV3l3OVBaVFd6L3ovbWFPL25saG5SYko3VE9mS3Njdkdldz0=' 'unsafe-eval'"). Source: ondrop attribute on INPUT element.  contacts

### PR DESCRIPTION
**Pull request by skjnldsv**
The contacts list will now flex between 250 and 350px.
I removed the column system as this is a terrible setup. I used a grid system, which should be adapted in the near futures with a propertygroup height calculation system and create a height unit based oon the number of rows an element takes (@Henni , tell me if I'm not understandable :p )

Fix #453
Fix #56

### CODE REVIEW

**Comment by Henni** [comment]
Not completely sure what you mean by "height calculation system". But grid layout seems to be a good fit for this layout.

Only possible issue I currently see is that this changes the flow of content from vertical to horizontal.
I'll test this out soon.

**Comment by skjnldsv** [comment]
Well, grid does not stack itself to prevent stuff like this:
capture d ecran_2018-02-14_13-23-01

So we have to define the number of rows each property will take

For example, an address element need to take 7 _heights units_ (6 inputs + 1 select) while a single mail property takes only two. This way we can force the items to be stacked together :)
This will require some calculation though, but I could not find a better way to do that :/

capture d ecran_2018-02-14_13-30-06

**Comment by skjnldsv** [comment]
Bump @Henni, we should merge and push 2.1.1 :)

**Comment by rollanders** [comment]
Confrim this solved the problem with narrow contactlist in Chrome

**Comment by juliushaertl** [comment]
I think the width of the grid layout should be limited, otherwise it will have quite huge space between the two columns on higher resolutions:

bildschirmfoto vom 2018-02-23 10-34-54

**Comment by skjnldsv** [comment]
@juliushaertl it's a temporary fix, the full grid layout will come pretty soon. :)

**Comment by skjnldsv** [comment]
@juliushaertl I decided to finally add the new mechanism in this pr. Can you try ? :)

**Comment by skjnldsv** [comment]
@Henni, the only downside is we need to edit our css to specify how many heights units (one input = 1 row) a property has. Most of them have only 1. The only I found was adresses, notes and detailed name. Do you see any other one?

@xh3n1 please update you rreview if you have the time :)

Example:
kazam

**Comment by skjnldsv** [comment]
Yup, looking really great! :)
capture d ecran_2018-02-26_11-58-18


@nextcloud/contacts please review!

File: css/_details.scss [hunk 1]
```diff
@@ -223,14 +223,30 @@ avatar.maximized .avatar-options > [class^="icon-"] {
...
-	section {
-		-webkit-column-count: 2; /* Chrome, Safari, Opera */
-		-moz-column-count: 2; /* Firefox */
-		column-count: 2;
+/* GRID */
+$grid-height-unit: 40px;
+$grid-input-padding: 7px;
+$grid-input-margin: 3px;
+$grid-column-width: 380px;
+$grid-input-height-with-margin: calc(#{$grid-height-unit} - (#{$grid-input-margin} * 2));
```
Comment (Henni) [inline css/_details.scss]: Calculations that are independent from the browser could be done in scss:
```scss
$grid-input-height-with-margin: #{$grid-height-unit - $grid-input-margin * 2};
```

File: .travis.yml [hunk 2]
```diff
@@ -28,8 +28,8 @@ matrix:
           env:  DB=pgsql CORE_BRANCH=stable13
         - php: 7.1
           env:  DB=pgsql CORE_BRANCH=stable13
-        - php: 7.2
-          env:  DB=pgsql CORE_BRANCH=stable13
+#        - php: 7.2
+#          env:  DB=pgsql CORE_BRANCH=stable13
```
Comment (Henni) [inline .travis.yml]: Why did you have to remove this?
Any idea how we could solve it?

Comment (skjnldsv) [reply .travis.yml]: I removed it for now . It's a known phpunit issue, I tried to solve it earlier, But my main focus was to have working tests again :)

We'll need to push a fix later on :)
https://github.com/sebastianbergmann/phpunit/issues/2453

File: css/_details.scss (hunk 1 above)
Comment (skjnldsv) [reply css/_details.scss]: Nice catch! :)

**Comment by juliushaertl** [comment]
Really nice @skjnldsv :+1:

**Pull request merged**